### ISSUE TITLE
feature: fluent scrolling

### ISSUE THREAD

**Issue by MicroCBer** [slint-ui/slint#4115]
Fluent scrolling takes an important part of the user experience, but Slint doesn't seem to support it.

I'd appreciate it if fluent scrolling could be implemented in standard ScrollView. To be honest, I had to implement the ScrollView again by myself because of the lack of this function.

[The demo here](https://slint.dev/releases/1.3.2/editor/?snippet=import+%7B+ScrollView+%2C+HorizontalBox%2C+GroupBox%2C+VerticalBox%7D+from+%22std-widgets.slint%22%3B%0Aexport+component+Example+inherits+Window+%7B%0A++++width%3A+200px%3B%0A++++height%3A+200px%3B%0A++++HorizontalBox+%7B%0A++++++++GroupBox+%7B%0A++++++++++++title%3A+%22Standard+implementation%22%3B%0A++++++++++++ScrollView+%7B%0A++++++++++++width%3A+200px%3B%0A++++++++++++height%3A+200px%3B%0A++++++++++++viewport-width%3A+300px%3B%0A++++++++++++viewport-height%3A+300px%3B%0A++++++++++++%0A++++++++++++Rectangle+%7B+width%3A+30px%3B+height%3A+30px%3B+x%3A+275px%3B+y%3A+50px%3B+background%3A+blue%3B+%7D%0A++++++++++++Rectangle+%7B+width%3A+30px%3B+height%3A+30px%3B+x%3A+175px%3B+y%3A+130px%3B+background%3A+red%3B+%7D%0A++++++++++++Rectangle+%7B+width%3A+30px%3B+height%3A+30px%3B+x%3A+25px%3B+y%3A+210px%3B+background%3A+yellow%3B+%7D%0A++++++++++++Rectangle+%7B+width%3A+30px%3B+height%3A+30px%3B+x%3A+98px%3B+y%3A+55px%3B+background%3A+orange%3B+%7D%0A++++++++%7D%0A++++++++%7D%0A++++++++GroupBox+%7B%0A++++++++title%3A+%22With+fluent+scroll%22%3B%0A++++++++Rectangle+%7B%0A++++++++++++clip%3A+true%3B%0A++++++++++++width%3A+200px%3B%0A++++++++++++height%3A+200px%3B%0A++++++++++++private+property+%3Clength%3E+dy%3A+0px%3B%0A++++++++++++touch%3A%3D+TouchArea+%7B%0A++++++++++++++++scroll-event+%28e%29%3D%3E+%7B%0A++++++++++++++++++++parent.dy+%3D+clamp%28e.delta-y+%2B+parent.dy%2C+-200px%2C+10px%29%3B%0A++++++++++++++++++++EventResult.accept%0A++++++++++++++++%7D%0A++++++++++++%7D%0A%0A++++++++++++Rectangle+%7B%0A++++++++++++%0A++++++++++++y%3A+10px+%2B+dy%3B%0A++++++++++++animate+y+%7B%0A+++++++++++++++++duration%3A+200ms%3B%0A+++++++++++++++++easing%3A+ease-in-out%3B%0A++++++++++++%7D%0A%0A++++++++++++%0A++++++++++++Rectangle+%7B+width%3A+30px%3B+height%3A+30px%3B+x%3A+275px%3B+y%3A+50px%3B+background%3A+blue%3B+%7D%0A++++++++++++Rectangle+%7B+width%3A+30px%3B+height%3A+30px%3B+x%3A+175px%3B+y%3A+130px%3B+background%3A+red%3B+%7D%0A++++++++++++Rectangle+%7B+width%3A+30px%3B+height%3A+30px%3B+x%3A+25px%3B+y%3A+210px%3B+background%3A+yellow%3B+%7D%0A++++++++++++Rectangle+%7B+width%3A+30px%3B+height%3A+30px%3B+x%3A+98px%3B+y%3A+55px%3B+background%3A+orange%3B+%7D%0A++++++++++++%7D%0A++++++++%7D%0A++++++++%7D%0A++++%7D%0A%7D%0A&style=native) shows the difference.

fortunately, it's pretty simple to implement it in slint thanks to your works :)

**Comment by ogoffart** [comment]
I assume this is about having an animation when using the mouse wheel.

This shouldn't be too hard to do by using `set_animated_value` instead of `set` around
https://github.com/slint-ui/slint/blob/a94148c12ced69fec92e3fdc99f068e4884f23c2/internal/core/items/flickable.rs#L316

**Comment by MicroCBer** [comment]
> I assume this is about having an animation when using the mouse wheel.
> 
> This shouldn't be too hard to do by using `set_animated_value` instead of `set` around
> https://github.com/slint-ui/slint/blob/a94148c12ced69fec92e3fdc99f068e4884f23c2/internal/core/items/flickable.rs#L316

nice! I don't know how to make properties animated on rust side so I failed to implement it after some investigations

**Comment by ogoffart** [comment]
>  I don't know how to make properties animated on rust side

Using the `Property::set_animated_value`  https://docs.rs/i-slint-core/latest/i_slint_core/struct.Property.html#method.set_animated_value

**Comment by MicroCBer** [comment]
> >  I don't know how to make properties animated on rust side
> 
> Using the `Property::set_animated_value`  https://docs.rs/i-slint-core/latest/i_slint_core/struct.Property.html#method.set_animated_value

Would you work on this? If not, I'll get the issue fixed and do pr when I have spare time.

**Comment by MicroCBer** [comment]
![image](https://github.com/user-attachments/assets/80879776-9618-4c08-90a8-4ba649a01553)
This is not working somehow, I'm I doing anything wrong? @ogoffart

**Comment by ogoffart** [comment]
That looks alright, what is not working?

Maybe if this is called every frame, the animation is always restarted and never progress (just an idea, i don't know what the problem is)

**Comment by MicroCBer** [comment]
> in new issue
> [ Report content ](/contact/report-content?content_url=https%3A%2F%2Fgithub.com%2Fslint-ui%2Fslint%2Fissues%2F4115%23issuecomment-2255401631&report=tronical+%28user%29)

Yes

**Comment by MicroCBer** [comment]
> How do you compute `animated`? Are you sure that it's true?

Yes

**Comment by MicroCBer** [comment]
> That looks alright, what is not working?
> 
> Maybe if this is called every frame, the animation is always restarted and never progress (just an idea, i don't know what the problem is)

I scroll, and nothing happens.

No, it's called only once each scroll.

Can you check what happened in my repo? I have no idea about this.

reproduce: cargo r --bin gallery, and check the standard list view
